src/app/weather-data.pipe.ts

```diff
@@ -0,0 +1,47 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({name: 'weatherData'})
+export class WeatherDataPipe implements PipeTransform {
+  transform(value: unknown, ...args: unknown[]): unknown {
+    return null;
+  }
+}
+
+@Pipe({name: 'windSpeed'})
+export class WindSpeedPipe implements PipeTransform {
+  transform(data: any):string{
+    return (data + "m/s");
+  }
+}
+
+@Pipe({name: 'pOPrecip'})
+export class pOPrecipPipe implements PipeTransform {
+  transform(data: any):string{
+    let ppRound = Math.round(data/5)*5;
+    if(ppRound >= 10){
+      ppRound = Math.round(ppRound/10)*10;
+    }
+    return (ppRound + "%");
+  }
+}
+
+@Pipe({name: 'vis'})
+export class visiblityPipe implements PipeTransform {
+  transform(data: any):string{
+    if(data<1000){
+      return ("Very Poor " +`(${data}m)`)
+    } else if(data<4000){
+      return ("Poor " +`(${data}m)`)
+    }else if(data<10000){
+      return ("Medium " +`(${data}m)`)
+    }else if(data<20000){
+      return ("Good " +`(${data}m)`)
+    }else if(data<40000){
+      return ("Very Good " +`(${data}m)`)
+    }else if(data>40000){
+      return ("Excellent " +`(${data}m)`)
+    }else{
+      return ("Something went wrong data is" + data);
+    }
```

Comment:
Too many exit points. This would be a reasonable bit of coding, but I'd do it a different way, something like this:

```suggestion
    const limits = [{value: 1000, prefix: 'Very Poor'}, {value: 4000, prefix: 'Poor'}, {value: 10000, prefix: 'Medium'},  {value: 20000, prefix: 'Good'}, {value: 40000, prefix: 'Very Good'}]
    for (const limit of limits){
       if(data < limit.value){
           return `${limit.prefix} (${data}m)`
       }
    } 
    if(data>40000){
      return (`Excellent (${data}m)`)
    }else{
      return (`Something went wrong, data is ${data}`);
    }
```

But I guess it's a coding style! 😄

Reply:
Im actually kind of annoyed that i didnt think of this myself prior :p especially since ive done similar before. and as a result i will be implementing this thankyou :)